File: competitive-events-api/src/main/java/es/jocasolo/competitiveeventsapi/dto/eventuser/EventUserPutDTO.java
package es.jocasolo.competitiveeventsapi.dto.eventuser;

import java.io.Serializable;

import es.jocasolo.competitiveeventsapi.dto.DTO;
import es.jocasolo.competitiveeventsapi.enums.eventuser.EventUserPrivilegeType;
import es.jocasolo.competitiveeventsapi.enums.eventuser.EventUserStatusType;
import lombok.Getter;
import lombok.Setter;

public class EventUserPutDTO extends DTO implements Serializable {

	private static final long serialVersionUID = 1L;

	@Getter
	@Setter
	private String username;

	@Getter
	@Setter
	private EventUserStatusType status;

	@Getter
	@Setter
	private EventUserPrivilegeType privilege;

	@Override
	public String toString() {
		return String.format("EventUserPostDTO [user=%s]", username);
	}

}

File: competitive-events-app/app/src/main/java/es/jocasolo/competitiveeventsapp/dto/eventuser/EventUserPutDTO.kt
package es.jocasolo.competitiveeventsapp.dto.eventuser

import com.google.gson.annotations.SerializedName
import es.jocasolo.competitiveeventsapp.dto.BackStackEntryDTO
import es.jocasolo.competitiveeventsapp.enums.eventuser.EventUserPrivilegeType
import es.jocasolo.competitiveeventsapp.enums.eventuser.EventUserStatusType

class EventUserPutDTO(
        @SerializedName("username") var username: String? = null,
        @SerializedName("status") var status: EventUserStatusType? = null,
        @SerializedName("privilege") var privilege: EventUserPrivilegeType? = null
) : BackStackEntryDTO() {

}